operations/app/src/environments/02-config/main.tf

```diff
@@ -39,9 +39,10 @@ module "key_vault" {
 }
 
 module "container_registry" {
-  source          = "../../modules/container_registry"
-  environment     = var.environment
-  resource_group  = var.resource_group
-  resource_prefix = var.resource_prefix
-  location        = var.location
+  source               = "../../modules/container_registry"
+  environment          = var.environment
+  resource_group       = var.resource_group
+  resource_prefix      = var.resource_prefix
+  location             = var.location
+  enable_content_trust = true
```

Comment:
Good call making this configurable.

Reply:
> Does the Container Registry need configuration to accept only signed code from specific certificates? The way I'm reading this now, it sounds like any valid signed container would work and not only containers signed with the specified CDC key.

To the best of my understanding, this setting just enables the Notary service on this registry. It does not control who can push up images, nor what gets pushed up, or which keys are used to sign things that are pushed up (if they are indeed signed).
With this setting on, you can start taking advantage of Content Trust, but you don't have to: even with this on, you can still push up unsigned images.

The next step is Azure-dependent and requires that we grant those (Service) principal that will do signing, permissions to sign and push signed containers (https://docs.microsoft.com/en-us/azure/container-registry/container-registry-content-trust#grant-image-signing-permissions) into the registry.

To the best of my understanding at this point in time, those are the only levers of control we have on the "making signed images appear in the registry"-side. I don't think that at this point, there is a mechanism to limit which keys are allowed to sign images. As long as the key is valid (and we have one of those), and the principal doing the signing is allowed to push up signed images, then images can be signed and will appear in the registry.

In terms of validation of signatures, my understanding is that the onus is on the client to do that validation (setting DOCKER_CONTENT_TRUST=1 when pulling enforces it, and will fail on attempting to pull unsigned images or images for which the signature is -cryptographically- invalid). I think the best way to think of these signed images is more akin to this: the fact that the signed image appears in the repo is a flag that says "yes you can trust this because only a legit, validated user of this repo (which includes "knowing the repository key") could have pushed this signed image up; if they weren't legit, then they couldn't have pushed up the signed image because the notary wouldn't have allowed this".